FV3GFS/FV3GFS_io.F90

```diff
@@ -2682,352 +2682,368 @@ subroutine phys_restart_write (IPD_Restart, Atm_block, Model, fv_domain, timesta
-    allocate (Diag_diag_manager_controlled(index)%data(nblks))
-    do nb = 1,nblks
-      Diag_diag_manager_controlled(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,4)
-    enddo
+      index = index + 1
+      Diag_diag_manager_controlled(index)%axes = 2
+      Diag_diag_manager_controlled(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_microphysics'
+      Diag_diag_manager_controlled(index)%desc = 'vertically integrated water vapor tendency due to microphysics'
+      Diag_diag_manager_controlled(index)%unit = 'kg/m**2/s'
+      Diag_diag_manager_controlled(index)%mod_name = 'gfs_phys'
+      Diag_diag_manager_controlled(index)%coarse_graining_method = 'area_weighted'
+      allocate (Diag_diag_manager_controlled(index)%data(nblks))
+      do nb = 1,nblks
+         Diag_diag_manager_controlled(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,4)
+      enddo
 
-    index = index + 1
-    Diag_diag_manager_controlled(index)%axes = 2
-    Diag_diag_manager_controlled(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_change_in_atmosphere_mass'
-    Diag_diag_manager_controlled(index)%desc = 'vertically integrated residual water vapor tendency'
-    Diag_diag_manager_controlled(index)%unit = 'kg/m**2/s'
-    Diag_diag_manager_controlled(index)%mod_name = 'gfs_phys'
-    Diag_diag_manager_controlled(index)%coarse_graining_method = 'area_weighted'
-    allocate (Diag_diag_manager_controlled(index)%data(nblks))
-    do nb = 1,nblks
-      Diag_diag_manager_controlled(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,5)
-   enddo
+      index = index + 1
+      Diag_diag_manager_controlled(index)%axes = 2
+      Diag_diag_manager_controlled(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_change_in_atmosphere_mass'
+      Diag_diag_manager_controlled(index)%desc = 'vertically integrated residual water vapor tendency'
+      Diag_diag_manager_controlled(index)%unit = 'kg/m**2/s'
+      Diag_diag_manager_controlled(index)%mod_name = 'gfs_phys'
+      Diag_diag_manager_controlled(index)%coarse_graining_method = 'area_weighted'
+      allocate (Diag_diag_manager_controlled(index)%data(nblks))
+      do nb = 1,nblks
+         Diag_diag_manager_controlled(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,5)
+      enddo
+
+      index = index + 1
+      Diag_diag_manager_controlled(index)%axes = 3
+      Diag_diag_manager_controlled(index)%name = 'co2'
+      Diag_diag_manager_controlled(index)%desc = 'carbon dioxide concentration'
+      Diag_diag_manager_controlled(index)%unit = 'volume mixing ratio'
+      Diag_diag_manager_controlled(index)%mod_name = 'gfs_phys'
+      Diag_diag_manager_controlled(index)%coarse_graining_method = MASS_WEIGHTED
+      allocate (Diag_diag_manager_controlled(index)%data(nblks))
+      do nb = 1,nblks
+         Diag_diag_manager_controlled(index)%data(nb)%var3 => IntDiag(nb)%co2(:,:)
+      enddo
```

Comment:
This is the only diagnostic field added; the other changes are related only to indentation.